lib/components/recordingSession/recorder.dart

```diff
@@ -28,94 +31,134 @@ class Recorder extends StatefulWidget {
 }
 
 class _RecorderState extends State<Recorder> {
-  MyWebSocket ws;
-  Directory directory;
   bool _isRecording = false;
-  StreamSubscription<RecordStatus> _recorderSubscription;
-  StreamSubscription<double> _dbPeakSubscription;
-  //StreamSubscription _playerSubscription;
-  FlutterSound flutterSound;
-  int userId;
-  String topic;
+
+  StreamSubscription<dynamic> _recorderSubscription;
+
+  FlutterSoundRecorder recorderModule;
+
   String _recorderTxt = '00:00:00';
   double _dbLevel;
-  int sessionCounter = 0;
 
   double sliderCurrentPosition = 0.0;
   double maxDuration = 1.0;
+  final Codec _codec = Codec.aacADTS;
+
+  int userId;
+  String topic;
+  MyWebSocket ws;
+  Directory tempDir;
+  String savePath;
+
+  int sessionCounter = 0;
+
+  Future<void> init() async {
+    tempDir = await getApplicationDocumentsDirectory();
+    recorderModule = FlutterSoundRecorder();
+  }
 
   @override
   void initState() {
     super.initState();
-    flutterSound = FlutterSound();
-    flutterSound.setSubscriptionDuration(0.01);
-    flutterSound.setDbPeakLevelUpdate(0.8);
-    flutterSound.setDbLevelEnabled(true);
+    init();
```

Comment:
method name isn't too self-explanatory, rename to something like intializeRecorder()